src/bootstrap/src/utils/cc_detect.rs

```diff
@@ -58,6 +59,11 @@ fn cc2ar(cc: &Path, target: TargetSelection, default_ar: PathBuf) -> Option<Path
     }
 }
 
+/// Creates and configures a new [`cc::Build`] instance for the given target.
+///
+/// This function initializes a new `cc::Build` object with a set of default options
+/// including optimization level, warning settings, and target/host configuration. It also
+/// applies static runtime configuration based on the target and build settings.
 fn new_cc_build(build: &Build, target: TargetSelection) -> cc::Build {
```

Comment:
Let's not mention about abstraction details as they can get outdated super easily.

```suggestion
/// Creates and configures a new [`cc::Build`] instance for the given target.
fn new_cc_build(build: &Build, target: TargetSelection) -> cc::Build {
```